numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -497,21 +498,33 @@ PyArray_SelectkindConverter(PyObject *obj, NPY_SELECTKIND *selectkind)
 static int searchside_parser(char const *str, Py_ssize_t length, void *data)
 {
     NPY_SEARCHSIDE *side = (NPY_SEARCHSIDE *)data;
+    int success = -1;
 
     if (length < 1) {
         return -1;
     }
     else if (str[0] == 'l' || str[0] == 'L') {
         *side = NPY_SEARCHLEFT;
-        return 0;
+        success = 0;
     }
     else if (str[0] == 'r' || str[0] == 'R') {
         *side = NPY_SEARCHRIGHT;
-        return 0;
-    }
-    else {
-        return -1;
+        success = 0;
+    }
+
+    /* Filters out the case sensitive/non-exact
+     * match inputs and other inputs and outputs DeprecationWarning */
+    if (!((length == 4 && strcmp(str, "left") == 0) ||
+          (length == 5 && strcmp(str, "right") == 0)) && success == 0) {
+        /* NumPy 1.19, 2020-04-21 */
+        if (DEPRECATE("inexact matches and case insensitive matches "
+                      "for search side are deprecated. Please use "
+                      "one of 'left' or 'right' instead") < 0) {
+            success = -1;
```

Comment:
You should check that `string_converter_helper` is actually able to handle exceptions being set. It might be replacing them, which it should not do.

Reply:
yes currently deprecation warnings surface as value errors. I will look into this.

Reply:
I have addressed this issue, I skip setting ValueError if error already occured.